src/ucs/debug/debug.c

```diff
@@ -498,15 +488,15 @@ static void ucs_debug_print_source_file(const char *file, unsigned line,
 
 static void ucs_debug_show_innermost_source_file(FILE *stream)
 {
-    const char *file, *function;
     struct backtrace bckt;
-    unsigned long address;
-    unsigned line;
-
-    ucs_debug_backtrace_create(&bckt);
-    while (backtrace_next(&bckt, &address, &file, &function, &line)) {
-        if (!ucs_debug_backtrace_is_excluded((void*)address, function)) {
-            ucs_debug_print_source_file(file, line, function, stream);
+    struct backtrace_line bckt_line;
+
+    ucs_debug_backtrace_create(&bckt, 0);
+    while (backtrace_next(&bckt, &bckt_line)) {
+        if (!ucs_debug_backtrace_is_excluded((void*)bckt_line.address,
```

Comment:
ucs_debug_backtrace_is_excluded() should only be called from backtrace_next() iterator

Reply:
removed